### PR TITLE
Resolve vulnerabilities

### PR DESCRIPTION
**Pull request by golubev-ivan**
**Task name**: root of the repo, AppStorePromote, AppStoreRelease, IpaResign

**Description**: Resolved vulnerabilities.

Updated dependencies in root of the repo:
- `minimist` to 1.2.5
- `mocha-junit-reporter` to 2.0.2
- `shelljs` to 0.8.5
- `validator` to 13.7.0
- `yargs` to 7.1.2

Updated dependencies in tasks:
- `azure-pipelines-task-lib` to 3.1.10

**Documentation changes required:** (Y/N) <Please mark if documentation changes are required>

**Added unit tests:** N

**Attached related issue:** N

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/blob/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,26 +13,26 @@
...
+    "shelljs": "^0.8.5",
     "sync-request": "4.1.0",
-    "tfx-cli": "^0.6.4",
-    "tslint": "^3.15.1",
+    "tfx-cli": "^0.11.0",
+    "tslint": "^5.20.1",
     "typescript": "3.2.2",
-    "validator": "3.33.0",
-    "yargs": "^6.4.0"
+    "validator": "13.7.0",
```
Comment (anatolybolshakov) [inline package.json:35]: This is quite huge leap from 3 to 13 major version - could you please double-check that there won't any changes required for this in tasks and behavior will still be the same?

Comment (DaniilShmelev) [reply package.json:35]: Yeah, double checked it - everything looks good. This package is only used in make-util. The signature of `validator.isLength` function has changed, but it has backwards compatibility fallback, so there is no need to change it:
```js
  if (_typeof(options) === 'object') {
    min = options.min || 0;
    max = options.max;
  } else {
    // backwards compatibility: isLength(str, min [, max])
    min = arguments[1] || 0;
    max = arguments[2];
  }
```

**Comment by golubev-ivan** [comment]
LGTM! Tested manually everything works fine.

**Review by DaniilShmelev — APPROVED** [review]

**Comment by DaniilShmelev** [comment]
@anatolybolshakov tested these cases:

Build scripts:
✔ make.js build
✔ make.js test
✔ gulp clean
✔ gulp build
✔ gulp test
✔ gulp packageprod
✔ gulp packagetest

Tasks:
✔ App Store Release, App Store Promote - released and promoted an app using service connection with fastlane data
✔ IPA Resign - resigned an app .ipa file

**Review by anatolybolshakov — APPROVED** [review]

**Review by mmrazik — APPROVED** [review]

**Pull request merged**